### PR TITLE
compatible batch copr with load-based replica read

### PR DESCRIPTION
**Pull request by you06**
pingcap/tidb#42322

Disable the server-is-busy retry when there are batched tasks, the application should handle it.

### CODE REVIEW

**Comment by disksing** [comment]
@you06 do you have anyone who can review this PR?

File: internal/locate/region_request.go [hunk 1]
```diff
@@ -961,6 +963,13 @@ func (s *replicaSelector) onServerIsBusy(bo *retry.Backoffer, ctx *RPCContext, s
 		ctx.Store.loadStats.Store(loadStats)
 
 		if s.busyThreshold != 0 {
+			// do not retry with batched coprocessor requests.
+			// it'll be region misses if we send the tasks to replica.
+			if req.Type == tikvrpc.CmdCop {
```
Comment (ekexium) [inline internal/locate/region_request.go:968]: Can we merge the nested if conditions into one?

File: internal/locate/region_request.go [hunk 2]
```diff
@@ -619,7 +619,9 @@ func (state *accessFollower) isCandidate(idx AccessIndex, replica *replica) bool
 			(!state.option.leaderOnly && (state.tryLeader || idx != state.leaderIdx) && replica.store.IsLabelsMatch(state.option.labels) && (!state.learnerOnly || replica.peer.Role == metapb.PeerRole_Learner)) &&
 				// And If the leader store is abnormal to be accessed under `ReplicaReadPreferLeader` mode, we should choose other valid followers
 				// as candidates to serve the Read request.
-				(!state.option.preferLeader || !replica.store.isSlow()))
+				(!state.option.preferLeader || !replica.store.isSlow())) &&
+		// If the stores are limited, check if the store is in the list.
+		replica.store.IsStoreMatch(state.option.stores)
```
Comment (ekexium) [inline internal/locate/region_request.go:624]: The single expression is scarily complicated... I'd prefer split them into multiple variables.

**Review by ekexium — APPROVED** [review]

**Review by cfzjywxk — APPROVED** [review]

**Pull request merged**